tests/integration/utils2.go

```diff
@@ -654,23 +655,38 @@ func restartNodes(
 		nodes[i] = n
 	}
 
+	// The index of the action after the last wait action before the current restart action.
+	// We wish to resume the wait clock from this point onwards.
+	waitGroupStartIndex := 0
+actionLoop:
+	for i := actionIndex; i >= 0; i-- {
+		switch testCase.Actions[i].(type) {
+		case WaitForSync:
+			// +1 as we do not wish to resume from the wait itself, but the next action
+			// following it. This may be the current restart action.
+			waitGroupStartIndex = i + 1
+			break actionLoop
+		}
+	}
+
 	syncChans := []chan struct{}{}
 	for _, tc := range testCase.Actions {
 		switch action := tc.(type) {
 		case ConnectPeers:
+			// Give the nodes a chance to connect to each other and learn about each other's subscribed topics.
+			time.Sleep(100 * time.Millisecond)
```

Comment:
nitpick: Perhaps store this as a constant somewhere at the top of this file?

Reply:
Really not a fan of that - it just means anyone reading has to break flow and scroll to the top of a file in order to view the value of this arbitrary value.